crates/core/src/node/p2p_impl.rs

```diff
@@ -149,33 +151,46 @@ impl NodeP2P {
 
         if let Some(query_target) = query_target {
             let joiner = self.op_manager.ring.connection_manager.own_location();
-            let msg = connect::ConnectMsg::Request {
-                id: tx,
-                target: query_target.clone(),
-                msg: connect::ConnectRequest::FindOptimalPeer {
-                    query_target,
-                    ideal_location,
-                    joiner,
-                    max_hops_to_live: self.op_manager.ring.max_hops_to_live,
-                    skip_connections: HashSet::new(),
-                    skip_forwards: HashSet::new(),
-                },
-            };
+            let ttl = self
+                .op_manager
+                .ring
+                .max_hops_to_live
+                .max(1)
+                .min(u8::MAX as usize) as u8;
+            let target_connections = self.op_manager.ring.connection_manager.min_connections;
+
+            let (tx, op, msg) = ConnectOp::initiate_join_request(
+                joiner,
+                query_target.clone(),
+                ideal_location,
+                ttl,
+                target_connections,
+            );
 
+            tracing::debug!(
+                %tx,
+                query_peer = %query_target.peer,
+                %ideal_location,
+                "Triggering connection maintenance connect request"
+            );
             self.op_manager
                 .notify_op_change(
-                    NetMessage::from(msg),
-                    OpEnum::Connect(Box::new(connect::ConnectOp::new(tx, None, None, None))),
+                    NetMessage::V1(NetMessageV1::Connect(msg)),
+                    OpEnum::Connect(Box::new(op)),
                 )
                 .await?;
         }
 
         Ok(())
     }
-    pub(super) async fn run_node(self) -> anyhow::Result<Infallible> {
+    pub(super) async fn run_node(mut self) -> anyhow::Result<Infallible> {
         if self.should_try_connect {
-            connect::initial_join_procedure(self.op_manager.clone(), &self.conn_manager.gateways)
-                .await?;
+            let join_handle = connect::initial_join_procedure(
```

Comment:
a better change even would be to do this in a structured way and listen to ther esult directly in the select below